Preserve CreationMeta when metafying views.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103152

This helps us avoid erroring / generate more accurate error messages
in Dynamo when doing mutations on views.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

ezyang committed:
Preserve CreationMeta when metafying views.

This helps us avoid erroring / generate more accurate error messages
in Dynamo when doing mutations on views.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

albanD commented on torch/csrc/autograd/init.cpp:
```diff
@@ -375,6 +383,22 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject* unused) {
     torch::autograd::PyDefaultSavedVariableHooks::pop_hooks();
   });
 
+  m.def("_is_differentiable_view", [](const at::Tensor& t) {
```
nit: is this one still needed? You use `t._is_view()` to check that

albanD reviewed (approved):
small nit but that sounds good!

ezyang committed:
Update on "Preserve CreationMeta when metafying views."


This helps us avoid erroring / generate more accurate error messages
in Dynamo when doing mutations on views.

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge

ezyang committed:
Update on "Preserve CreationMeta when metafying views."


This helps us avoid erroring / generate more accurate error messages
in Dynamo when doing mutations on views.

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge -f "asan timeout seems unrelated"
